yaml/builders/image-test.yaml

```diff
@@ -8,6 +8,17 @@
             #!/bin/bash
             set -ex
 
+            # Generate sources, commit changes into generated branch and show diff
+            ssh -F ssh_config host 'set -ex; \
+              cd sources; GIT_HASH=$(git rev-parse HEAD); \
+              git config user.name "SCLorg Jenkins"; git config user.email "sclorg@redhat.com"; \
+              if git checkout generated; then \
+                git submodule update; \
+                result="$(./update $GIT_HASH)"; \
+                [ "$result" = "Nothing changed" ] || git diff generated~1 generated -- $(sed -n 's/^VERSIONS[[:space:]]*=//p' Makefile); \
+                git checkout -f $GIT_HASH && git submodule update; \
+              fi;'
```

Comment:
This is executed for every run of jenkins, right?  Even for the pull requests?

Reply:
Yes (builds and PR testing share the same part - running tests for the image). But it is not pushed anywhere... Also git repository is freshly cloned for every run, so it should be "safe".

Also we want to have diff shown even for PRs, so we need to do something like this. Or any other idea how to do it?

Reply:
Correct, I just wanted to be sure that when **this** code fails (for any reason) we'll see it in asap (before merging the PR), so any failure here means failure of the whole jenkins job?

Reply:
> so any failure here means failure of the whole jenkins job

Right.